configuration_go/abstr/kubernetes/thanos/compactor/compactor.go

```diff
@@ -1 +1,168 @@
+	TracingConfig                      string        `opt:"tracing.config"`
+	TracingConfigFile                  string        `opt:"tracing.config-file"`
+	Version                            bool          `opt:"version,noval"`
+	Wait                               bool          `opt:"wait,noval"`
+	WaitInterval                       time.Duration `opt:"wait-interval"`
+	WebDisable                         bool          `opt:"web.disable"`
+	WebDisableCors                     bool          `opt:"web.disable-cors"`
+	WebExternalPrefix                  string        `opt:"web.external-prefix"`
+	WebPrefixHeader                    string        `opt:"web.prefix-header"`
+	WebRoutePrefix                     string        `opt:"web.route-prefix"`
+
+	// Extra options not officially supported by the compactor.
+	cmdopt.ExtraOpts
+}
+
+// DefaultMetaConfig returns the default meta configuration for the compactor.
+func DefaultMetaConfig() k8sutilv2.MetaConfig {
+	return k8sutilv2.MetaConfig{
+		Name:      "observatorium-thanos-compact",
+		Namespace: defaultNamespace,
+		Labels: map[string]string{
+			k8sutil.NameLabel:      "thanos-compact",
+			k8sutil.InstanceLabel:  "observatorium",
+			k8sutil.PartOfLabel:    "observatorium",
+			k8sutil.ComponentLabel: "database-compactor",
+			k8sutil.VersionLabel:   defaultImageTag,
+		},
+	}
+}
+
+// NewBaseContainerProvider returns a new container provider for the compactor.
+func NewBaseContainerProvider(c *CompactorOptions) *k8sutilv2.Container {
+	if c == nil {
+		c = &CompactorOptions{}
+	}
+
+	httpPort := defaultHTTPPort
+	if c.HttpAddress.Port != 0 {
+		httpPort = c.HttpAddress.Port
+	}
+
+	livenessProbeCfg := k8sutilv2.ProbeConfig{
+		FailureThreshold: 4,
+		PeriodSeconds:    30,
+	}
+	readinessProbeCfg := k8sutilv2.ProbeConfig{
+		FailureThreshold: 20,
+		PeriodSeconds:    5,
+	}
+
+	return &k8sutilv2.Container{
+		Image:           defaultImage,
+		ImageTag:        defaultImageTag,
+		Name:            "thanos",
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		Args:            append([]string{"compact"}, cmdopt.GetOpts(c)...),
+		Ports: []corev1.ContainerPort{
+			{
+				Name:          "http",
+				ContainerPort: int32(httpPort),
+				Protocol:      corev1.ProtocolTCP,
+			},
+		},
+		LivenessProbe:  k8sutilv2.NewProbe("/-/healthy", httpPort, livenessProbeCfg),
+		ReadinessProbe: k8sutilv2.NewProbe("/-/ready", httpPort, readinessProbeCfg),
+		Resources:      k8sutilv2.NewResourcesRequirements("2", "3", "2000Mi", "3000Mi"),
+
+		ServicePorts: []corev1.ServicePort{
+			k8sutilv2.NewServicePort("http", httpPort, httpPort),
+		},
+		MonitorPorts: []monv1.Endpoint{
+			{
+				Port:           "http",
+				RelabelConfigs: k8sutilv2.GetDefaultServiceMonitorRelabelConfig(),
+			},
+		},
+	}
+}
+
+// NewStatefulSet returns a new statefulset container provider for the compactor.
+// It includes the compactor container and the volume claims.
+func NewSSContainerProvider(c *CompactorOptions) *k8sutilv2.Container {
```

Comment:
This returns a new **statefulset** **container**? 🤔 

Shouldn't it be only statefulset or only container? Of course a statefulset has pods, that has containers.

I am missing a high-order function to create the Compactor here. Something that creates the `compactorStatefulSet` variable like in there:  https://github.com/observatorium/observatorium/pull/537/files#diff-cd4defa334f8d951bb98d5d340bd8afb7901fc72a89fcf8fa8bd5888f6666fc7R314-R337

Reply:
The idea was to provide the possibility to deploy the compactor either as a statefulSet or a Deployment (not sure it make sense). Providing more flexibility for the lib user.
But, yes I'll add a default StatefulSet returned by the compactor, and probably remove this function. It seems to align more with what is expected from this lib and @saswatamcode also expects that.

Reply:
This function has been removed.